packages/dds/tree/src/simple-tree/api/customTree.ts

```diff
@@ -36,24 +36,56 @@ import { isObjectNodeSchema } from "../node-kinds/index.js";
 /**
  * Options for how to interpret or encode a tree when schema information is available.
  * @alpha
+ * @input
  */
-export interface TreeEncodingOptions {
+export interface TreeEncodingOptions<TKeyOptions = KeyEncodingOptions> {
```

Comment:
What is the practical benefit to making the key options a generic type parameter?

Reply:
Some use cases, like import and json schema export don't support all the options, so this lets us provide compile time restrictions to match the runtime limitations without having to duplicate this interface.